### ISSUE TITLE
GCC in packageOverrides

### ISSUE THREAD

**Issue by vcunat** [NixOS/nixpkgs#2817]
Unresolved discussion on ML: https://www.mail-archive.com/nix-dev@lists.science.uu.nl/msg11889.html

```
  packageOverrides = pkgs_: with pkgs_; {
    testgcc2 = pkgs_.gcc;
  };
```

```
% nix-instantiate '<nixpkgs>' -A gcc
warning: you did not specify `--add-root'; the result might be removed by the garbage collector
/nix/store/993glmz6hxbld0ry81jwv7c3z7cgaz68-gcc-wrapper-4.8.2.drv

% nix-instantiate '<nixpkgs>' -A testgcc2
warning: you did not specify `--add-root'; the result might be removed by the garbage collector
/nix/store/41nacbfpr8jzsk3k03jjlqb2r2xis5gf-gcc-wrapper-4.8.2.drv
```

Note that regular packages in packageOverrides do NOT cause such problems.

**Comment by bennofs** [comment]
I'll try to track this down. Always wanted to learn about how stdenv and cross building works, so this looks like a good opportunity to dig into the code and understand how it works.

**Comment by zimbatm** [comment]
Fixed by #9400
